google/resource_compute_instance.go

```diff
@@ -1501,8 +1541,18 @@ func expandBootDisk(d *schema.ResourceData, config *Config, zone *compute.Zone,
 	}
 
 	if v, ok := d.GetOk("boot_disk.0.disk_encryption_key_raw"); ok {
-		disk.DiskEncryptionKey = &computeBeta.CustomerEncryptionKey{
-			RawKey: v.(string),
+		if v != "" {
```

Comment:
Just curious, did you run into any cases where d.GetOk returned true, but the value was empty?

Reply:
(I ask because that should be impossible, and so the code could be a tiny bit cleaner without the extra `!= ""` checks)

Reply:
Without the check the API was complaining that I was trying to create a disk with a customer-supplied encryption key without specifying a key.

Reply:
Hmm that's surprising. I don't think it's worth either of our time to really dig in though since it seems to work just fine with the checks. Thanks!